python/tk_nuke_writenode/handler.py

```diff
@@ -1654,8 +1658,17 @@ def __gather_render_settings(self, node, is_proxy=False):
             # check for 'channel' for backwards compatibility
             if "output" in render_template.keys or "channel" in render_template.keys:
                 output_name = node.knob(TankWriteNodeHandler.OUTPUT_KNOB_NAME).value()
-            
-        return (render_template, width, height, output_name)
+
+        # extension = node.knob('tk_file_type').value()
+
+        # decode extension from profile['file_type']
+        # this feels like a bad way because __apply_cached_file_format_settings()
+        # resets render_path (which is not cached) and extension reverts to default value
+        profile_name = node.knob('profile_name').value()
+        profile = self._profiles.get(profile_name, {})
+        extension = profile.get('file_type')
+
```

Comment:
Assumes file_type in the profile contains the extension to be used in the file name (eg. jpg) rather than the one expected by Nuke (eg. jpeg) which is stored in the tk_file_type knob. Can instead do a reverse-lookup on the extension choices dictionary (search by value for key) but that feels inefficient.